### PR TITLE
Translation for zh-tw, zh-hk are added; zh-cn is optimized.

### PR DESCRIPTION
**Pull request by thomas930**
Translation of the following languages are added: 
- zh-hk - Traditional Chinese - Hong Kong
- zh-tw - Traditional Chinese - Taiwan 

Translation for the following language is optimized: 
- zh-cn - Simplified Chinese

All Translatons are referenceing on latest english translation files (M3UAndroid/i18n/src/main/res/values).

### CODE REVIEW

**Comment by oxyroid** [comment]
Are you a Chinese speaker? Some adjustments are not correct.

**Comment by thomas930** [comment]
> Are you a Chinese speaker? Some adjustments are not correct.

Yes. My mother tongue is Chinese, familar with traditional chinese. 

Please indicate some examples of incorrect adjustment.

File: i18n/src/main/res/values-zh-rCN/feat_playlist_configuration.xml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <string name="feat_playlist_configuration_title">标题</string>
-    <string name="feat_playlist_configuration_user_agent">用户代理（UA）</string>
-    <string name="feat_playlist_configuration_enabled_epgs">启用的 EPG</string>
+    <string name="feat_playlist_configuration_user_agent">用户代理（UA)</string>
+    <string name="feat_playlist_configuration_enabled_epgs">己启用电子节目指南 EPG</string>
```
Comment (oxyroid) [inline i18n/src/main/res/values-zh-rCN/feat_playlist_configuration.xml:5]: 已 but not 己，and "EPG" can be removed

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml [hunk 2]
```diff
@@ -2,27 +2,24 @@
...
     <string name="feat_setting_sync_mode">同步策略</string>
-
     <string name="feat_setting_playlist_management">订阅管理</string>
     <string name="feat_setting_connect_timeout">最大超时时间</string>
     <string name="feat_setting_god_mode">上帝模式</string>
-    <string name="feat_setting_god_mode_description">通过物理按键调节界面</string>
-    <string name="feat_setting_experimental_mode">试验特性</string>
-    <string name="feat_setting_experimental_mode_description">不稳定的功能可能引发致命错误</string>
+    <string name="feat_setting_god_mode_description">通过实体音量键调节界面</string>
+    <string name="feat_setting_experimental_mode">试验模式</string>
```
Comment (oxyroid) [inline i18n/src/main/res/values-zh-rCN/feat_setting.xml:18]: incorrect

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml [hunk 3]
```diff
@@ -32,37 +29,35 @@
     <string name="feat_setting_clip_mode_clip">裁剪</string>
     <string name="feat_setting_clip_mode_stretched">拉伸</string>
     <string name="feat_setting_auto_refresh_channels">自动刷新频道</string>
-    <string name="feat_setting_auto_refresh_channels_description">进入播放列表自动刷新</string>
-
-    <string name="feat_setting_full_info_player">详细信息的播放器</string>
-    <string name="feat_setting_full_info_player_description">播放器中展示更多的信息</string>
+    <string name="feat_setting_auto_refresh_channels_description">进入频道时自动刷新</string>
```
Comment (oxyroid) [inline i18n/src/main/res/values-zh-rCN/feat_setting.xml:32]: 频道 => Channel
播放列表/订阅 => Playlist

This feature means refreshing channels once entering a playlist

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml [hunk 4]
```diff
@@ -32,37 +29,35 @@
...
+    <string name="feat_setting_local_storage">本地文件</string>
+    <string name="feat_setting_error_unselected_file">未有选择文件</string>
     <string name="feat_setting_error_blank_url">目标地址为空</string>
     <string name="feat_setting_use_dynamic_colors">动态配色</string>
-    <string name="feat_setting_use_dynamic_colors_unavailable">仅在 Android 12 及更高版本可用</string>
+    <string name="feat_setting_use_dynamic_colors_unavailable">适用于 Android 12 或以上版本</string>
     <string name="feat_setting_colorful_background">炫彩背景</string>
     <string name="feat_setting_restore_schemes">恢复主题</string>
-    <string name="feat_setting_not_implementation">此功能当前版本不可用</string>
+    <string name="feat_setting_not_implementation">当前版本不支援此功能</string>
```
Comment (oxyroid) [inline i18n/src/main/res/values-zh-rCN/feat_setting.xml:51]: 支援 means healing in game(like Overwatch?

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml [hunk 5]
```diff
@@ -72,58 +67,66 @@
...
     <string name="feat_setting_always_replay">总是显示重播按钮</string>
-    <string name="feat_setting_player_panel">播放器控制台</string>
-    <string name="feat_setting_player_panel_description">竖屏状态下上滑播放器</string>
+    
+    <string name="feat_setting_player_panel">播放器面板</string>
+    <string name="feat_setting_player_panel_description">在竖屏状态时向上滑动以将其展开</string>
 
-    <string name="feat_setting_cache">边播边存</string>
-    <string name="feat_setting_cache_description">此选项可能导致无法正常播放</string>
+    <string name="feat_setting_cache">播放时缓存</string>
```
Comment (oxyroid) [inline i18n/src/main/res/values-zh-rCN/feat_setting.xml:113]: restore

File: i18n/src/main/res/values-zh-rCN/feat_stream.xml [hunk 6]
```diff
@@ -4,18 +4,22 @@
...
+    
     <string name="feat_channel_tooltip_on_back_pressed">返回</string>
     <string name="feat_channel_tooltip_mute">静音</string>
     <string name="feat_channel_tooltip_unmute">取消静音</string>
     <string name="feat_channel_tooltip_favourite">收藏</string>
     <string name="feat_channel_tooltip_unfavourite">取消收藏</string>
     <string name="feat_channel_tooltip_download">下载</string>
     <string name="feat_channel_tooltip_stop_download">停止下载</string>
-    <string name="feat_channel_tooltip_cast">投屏</string>
+    <string name="feat_channel_tooltip_cast">投屏 Cast</string>
```
Comment (oxyroid) [inline i18n/src/main/res/values-zh-rCN/feat_stream.xml:15]: 🤔

**Review by oxyroid — COMMENTED** [review]
I have added some comments

File: i18n/src/main/res/values-zh-rCN/feat_playlist_configuration.xml (hunk 1 above)
Comment (thomas930) [reply i18n/src/main/res/values-zh-rCN/feat_playlist_configuration.xml]: solved by fix 已 typo error and remove "EPG" wordings.

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml (hunk 2 above)
Comment (thomas930) [reply i18n/src/main/res/values-zh-rCN/feat_setting.xml:18]:
```
<string name="feat_setting_god_mode_description">adjust layouts by physical volume buttons</string>
    <string name="feat_setting_experimental_mode">experimental mode</string>
    <string name="feat_setting_experimental_mode_description">unstable features can throw fatal errors</string>
```
I just translate by these english wordings. Please advise or modify english version if needed.

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml (hunk 3 above)
Comment (thomas930) [reply i18n/src/main/res/values-zh-rCN/feat_setting.xml:32]: `<string name="feat_setting_auto_refresh_channels_description">automatically refreshed when enter the channel</string>
`
I just translate by these english wordings. Please advise or modify english version if needed.

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml (hunk 4 above)
Comment (thomas930) [reply i18n/src/main/res/values-zh-rCN/feat_setting.xml]: "支援／不支援某些功能” is commonly used in Hong Kong or Taiwan, maybe I revert zh-cn wordings.

File: i18n/src/main/res/values-zh-rCN/feat_setting.xml (hunk 5 above)
Comment (thomas930) [reply i18n/src/main/res/values-zh-rCN/feat_setting.xml:113]:
```
<string name="feat_setting_cache">cache while playing</string>
    <string name="feat_setting_cache_description">this option may prevent normal playback</string>
    <string name="feat_setting_clear_cache">clear cache</string>
```
I just translate by these english wordings. Please advise or modify english version if needed.

Comment (oxyroid) [reply i18n/src/main/res/values-zh-rCN/feat_setting.xml:113]: You can modify the English wordings.

File: i18n/src/main/res/values-zh-rCN/feat_stream.xml (hunk 6 above)
Comment (oxyroid) [reply i18n/src/main/res/values-zh-rCN/feat_stream.xml]: You'd better revert all mainland translation if you do not learn more about it.

Comment (thomas930) [reply i18n/src/main/res/values-zh-rCN/feat_stream.xml]: @oxyroid 
It's my first time to see a project leader is impolite and irresponsible manner in GitHub. 
I use a whole day around 12 hours to work on 3 languages (dialects) translations, and feedback is "modify the original English wordings" and "revert all things". Asking me either do all things or do nothing. 

Okay, I will close PR and NEVER contribute again any more on your project. 

**Shame on you!**

**Comment by thomas930** [comment]
@oxyroid
It's my first time to see a project leader is impolite and irresponsible manner in GitHub.
I use a whole day around 12 hours to work on 3 languages (dialects) translations, and feedback is "modify the original English wordings" and "revert all things". Asking me either do all things or do nothing.

Okay, I will close PR and NEVER contribute again any more on your project.

Shame on you!

**Pull request closed without merging**